core/src/test/java/org/apache/calcite/sql2rel/RelFieldTrimmerTest.java

```diff
@@ -183,4 +190,96 @@
     assertThat(trimmed, hasTree(expected));
   }
 
+  @Test public void testCalcFieldTrimmer0() {
+    final RelBuilder builder = RelBuilder.create(config().build());
+    final RelNode root =
+        builder.scan("EMP")
+            .project(builder.field("EMPNO"), builder.field("ENAME"), builder.field("DEPTNO"))
+            .exchange(RelDistributions.SINGLETON)
+            .project(builder.field("EMPNO"), builder.field("ENAME"))
+            .build();
+
```

Comment:
What's the difference between `testCalcFieldTrimmer0 ` and `testCalcFieldTrimmer1`?

Reply:
`testCalcFieldTrimmer0` is used for check Project operator.
`testCalcFieldTrimmer1` is used for check Filter operator.